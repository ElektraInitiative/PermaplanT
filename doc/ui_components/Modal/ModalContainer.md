# ModalContainer

## Description

A container component for different kinds of modals.

## Props

| Name     | Type      | Optional | Description                                                  |
| -------- | --------- | -------- | ------------------------------------------------------------ |
| children | ReactNode | no       | The modals associated with this container.                   |
| show     | boolean   | no       | Children will only be displayed if this prop is set to true. |
