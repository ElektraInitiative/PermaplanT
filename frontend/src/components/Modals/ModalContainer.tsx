import { AnimatePresence, motion } from 'framer-motion';

interface ModalContainerProps {
  children: React.ReactNode;
  show: boolean;
}

export default function ModalContainer({ children, show }: ModalContainerProps) {
  return (
    <AnimatePresence>
      {show && (
        <motion.div
          className="fixed top-[15%] left-[50%] z-[1020]"
          initial={{ opacity: 0, translateX: '-50%' }}
          animate={{
            opacity: 100,
            transition: { delay: 0, duration: 0.1 },
          }}
          exit={{
            opacity: 0,
            transition: { delay: 0, duration: 0.1 },
          }}
        >
          {children}
        </motion.div>
      )}
    </AnimatePresence>
  );
}
