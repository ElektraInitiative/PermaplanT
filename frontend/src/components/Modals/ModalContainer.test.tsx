import ModalContainer from './ModalContainer';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ModalContainer show={true}> </ModalContainer>
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
