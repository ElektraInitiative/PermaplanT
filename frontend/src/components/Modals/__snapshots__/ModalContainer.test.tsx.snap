// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fixed left-[50%] top-[15%] z-[1020]"
  style={
    {
      "opacity": 0,
      "transform": "translateX(-50%) translateZ(0)",
    }
  }
>
   
</div>
`;
